### ISSUE TITLE
[Windows] CommandBar throws an unhandled exception using Material resources

### ISSUE THREAD

**Issue by FrozDark** [unoplatform/uno.toolkit.ui#288]
## Current behavior
Navigating to a page with CommandBar crashes the windows app with unhandled exception using one of these resources
```
<MaterialResources xmlns="using:Uno.Material" />
<ToolkitResources xmlns="using:Uno.Toolkit.UI" />
<MaterialToolkitResources xmlns="using:Uno.Toolkit.UI.Material" />
```

## Expected behavior
No crash

## How to reproduce it (as minimally and precisely as possible)
1. Create uno project
2. Add toolkit with material resources
3. Add resources to the Dictionary
4. Create page with CommandBar
5. Start windows and navigate to the page containing CommandBar

## Environment

Nuget Package:
Uno.WinUI 4.5.0-dev.521
Uno.Toolkit.WinUI 2.3.0-dev.6
Uno.Toolkit.WinUI.Material 2.3.0-dev.6

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [X] Windows
- [ ] Build tasks

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [X] 2022 (version: )
- [ ] for Mac (version: )

**Comment by kazo0** [comment]
Thanks for the report @FrozDark, I will take a look.

Just to be sure, are you initializing the MaterialColors in your app.xaml prior to the MaterialResources init? Similar to Step #3 here: https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html#getting-started

It could be that the color resources aren't being resolved. Otherwise, it is very possible there's a bug here and I will investigate, thanks!

**Comment by FrozDark** [comment]
> Thanks for the report @FrozDark, I will take a look.
> 
> Just to be sure, are you initializing the MaterialColors in your `App.xaml` prior to the MaterialResources init? Similar to Step 3 here: https://platform.uno/docs/articles/external/uno.themes/doc/material-getting-started.html#getting-started
> 
> It could be that the color resources aren't being resolved. Otherwise, it is very possible there's a bug here and I will investigate, thanks!

Yes. MaterialColors resource is initialized first. Screenshot here if it helps
![ujt3](https://user-images.githubusercontent.com/6670150/185141797-88d72ad5-8753-4b6a-bc1f-bde1d6d3eb32.jpg)

Also I made mistake writing that it throws UnhandledException. To correct myself it throws "System.Runtime.InteropServices.COMException" in WinRT.Runtime.dll

**Comment by FrozDark** [comment]
Attached the repro
[UnoAppTest.zip](https://github.com/unoplatform/uno.toolkit.ui/files/9362960/UnoAppTest.zip)

**Comment by kazo0** [comment]
Ah okay then yes, definitely an issue. Thanks for the repro!
